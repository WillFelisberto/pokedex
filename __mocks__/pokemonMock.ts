import { PokemonProps } from '@/components/organisms/PokeCard';

export const mockPokemon: PokemonProps = {
  id: 2,
  name: 'ivysaur',
  sprites: {
    front_default:
      'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/2.png'
  },
  types: ['grass', 'poison'],
  category: 'ivysaur',
  weight: 130,
  height: 10,
  weaknesses: ['flying', 'poison', 'bug', 'fire', 'ice'],
  evolutions: [
    {
      id: 1,
      name: 'bulbasaur',
      url: 'https://pokeapi.co/api/v2/pokemon/1/',
      sprite: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png',
      types: ['grass', 'poison'],
      level: null
    },
    {
      id: 2,
      name: 'ivysaur',
      url: 'https://pokeapi.co/api/v2/pokemon/2/',
      sprite: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png',
      types: ['grass', 'poison'],
      level: 16
    },
    {
      id: 3,
      name: 'venusaur',
      url: 'https://pokeapi.co/api/v2/pokemon/3/',
      sprite: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/3.png',
      types: ['grass', 'poison'],
      level: 32
    }
  ],
  abilities: [
    { name: 'overgrow', is_hidden: false },
    { name: 'chlorophyll', is_hidden: true }
  ],
  genderRatio: {
    male: 87.5,
    female: 12.5
  }
};
