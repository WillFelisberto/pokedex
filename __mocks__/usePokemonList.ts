/* eslint-disable no-console */
// src/hooks/__mocks__/usePokemonList.ts
export const usePokemonList = () => ({
  pokemons: [
    {
      id: 1,
      name: 'Pikachu',
      image: '/images/pikachu.png',
      type: 'Electric'
    },
    {
      id: 2,
      name: 'Charmander',
      image: '/images/charmander.png',
      type: 'Fire'
    }
  ],
  isFetching: false,
  totalPages: 5,
  currentPage: 1,
  goToPage: (page: number) => console.log(`Going to page ${page}`)
});
