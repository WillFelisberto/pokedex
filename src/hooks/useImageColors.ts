import { useBackgroundColor } from '@/context/BackgroundColorContext';
import { Vibrant } from 'node-vibrant/browser';
import { useState, useEffect } from 'react';

export const useImageColors = (imageUrl: string) => {
  const { setBackgroundColor } = useBackgroundColor();
  const [colors, setColors] = useState<{
    vibrant?: string;
    muted?: string;
    darkVibrant?: string;
    darkMuted?: string;
    lightVibrant?: string;
    lightMuted?: string;
  }>({});

  const [isLoading, setIsLoading] = useState(true); // Estado de carregamento

  useEffect(() => {
    if (!imageUrl) return;

    const fetchColors = async () => {
      setIsLoading(true); // Inicia o loading antes de buscar as cores
      try {
        const palette = await Vibrant.from(imageUrl).getPalette();
        const extractedColor = palette.DarkMuted?.hex || palette.Muted?.hex || '#ffffff';

        setColors({
          vibrant: palette.Vibrant?.hex,
          muted: palette.Muted?.hex,
          darkVibrant: palette.DarkVibrant?.hex,
          darkMuted: palette.DarkMuted?.hex,
          lightVibrant: palette.LightVibrant?.hex,
          lightMuted: palette.LightMuted?.hex
        });

        setBackgroundColor(extractedColor); // Atualiza o fundo dinamicamente
      } catch (error) {
        console.error('Erro ao extrair cores:', error);
      } finally {
        setIsLoading(false); // Finaliza o loading após a execução
      }
    };

    fetchColors();
  }, [imageUrl, setBackgroundColor]);

  return { colors, isLoading };
};
