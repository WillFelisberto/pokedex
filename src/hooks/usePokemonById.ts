/* eslint-disable @typescript-eslint/no-explicit-any */
import { useQuery } from '@tanstack/react-query';

const API_URL = process.env.NEXT_PUBLIC_POKEAPI_URL;

const generationToRegionMap: Record<string, string> = {
  'generation-i': 'Kanto',
  'generation-ii': 'Johto',
  'generation-iii': 'Hoenn',
  'generation-iv': 'Sinnoh',
  'generation-v': 'Unova',
  'generation-vi': 'Kalos',
  'generation-vii': 'Alola',
  'generation-viii': 'Galar',
  'generation-ix': 'Paldea'
};

const getIdFromSpeciesUrl = (url: string): number => {
  const parts = url.split('/');

  return parseInt(parts[parts.length - 2], 10);
};

const getPokemonDetailsById = async (id: number) => {
  const res = await fetch(`${API_URL}/pokemon/${id}`);
  const details = await res.json();

  return {
    id: details.id,
    name: details.name,
    weight: details.weight,
    height: details.height,
    sprite: details.sprites.other['official-artwork'].front_default,
    evolutionSprite: details.sprites.front_default,
    types: details.types.map((t: any) => t.type.name),
    abilities: details.abilities.map((a: any) => ({
      name: a.ability.name,
      is_hidden: a.is_hidden
    })),
    speciesUrl: details.species.url,
    stats: details.stats
  };
};

const processEvolutionNode = async (
  node: any,
  evolutionLevel: number | null = null
): Promise<
  Array<{
    id: number;
    name: string;
    sprite: string;
    types: string[];
    level: number | null;
  }>
> => {
  const id = getIdFromSpeciesUrl(node.species.url);
  const details = await getPokemonDetailsById(id);

  const currentEvolution = {
    id: details.id,
    name: node.species.name,
    sprite: details.evolutionSprite,
    types: details.types,
    level: evolutionLevel
  };

  let evolutions = [currentEvolution];

  for (const child of node.evolves_to) {
    const childEvolutionLevel = child.evolution_details?.[0]?.min_level ?? null;
    const childEvolutions = await processEvolutionNode(child, childEvolutionLevel);
    evolutions = evolutions.concat(childEvolutions);
  }

  return evolutions;
};

const fetchPokemonById = async (id: number) => {
  const details = await getPokemonDetailsById(id);

  // Buscar fraquezas
  const weaknessesRes = await fetch(`${API_URL}/type/${details.types[0]}`);
  const weaknessesData = await weaknessesRes.json();
  const weaknesses = weaknessesData.damage_relations.double_damage_from.map((w: any) => w.name);

  // Buscar informações de evolução
  const speciesRes = await fetch(details.speciesUrl);
  const speciesData = await speciesRes.json();

  // Pegando o nome em japonês da lista de nomes
  const japaneseNameEntry = speciesData.names.find((n: any) => n.language.name === 'ja');
  const japaneseName = japaneseNameEntry ? japaneseNameEntry.name : 'Desconhecido';

  // Identificando a geração e mapeando para a região correspondente
  const generationName = speciesData.generation.name;
  const region = generationToRegionMap[generationName] || 'Desconhecida';

  // Pegando os base stats
  const baseStats = {
    hp: details.stats.find((s: any) => s.stat.name === 'hp')?.base_stat || 0,
    attack: details.stats.find((s: any) => s.stat.name === 'attack')?.base_stat || 0,
    defense: details.stats.find((s: any) => s.stat.name === 'defense')?.base_stat || 0,
    specialAttack: details.stats.find((s: any) => s.stat.name === 'special-attack')?.base_stat || 0,
    specialDefense:
      details.stats.find((s: any) => s.stat.name === 'special-defense')?.base_stat || 0,
    speed: details.stats.find((s: any) => s.stat.name === 'speed')?.base_stat || 0
  };

  const evolutionRes = await fetch(speciesData.evolution_chain.url);
  const evolutionData = await evolutionRes.json();

  const evolutionChain = await processEvolutionNode(evolutionData.chain);

  // Buscar distribuição de gênero
  const genderRes = await fetch(`${API_URL}/pokemon-species/${details.id}/`);
  const genderData = await genderRes.json();
  const maleRatio = 100 - genderData.gender_rate * 12.5;
  const femaleRatio = genderData.gender_rate * 12.5;

  return {
    id: details.id,
    name: details.name,
    japaneseName, // Adicionando nome japonês
    region,
    baseStats,
    weight: details.weight,
    height: details.height,
    sprite: details.sprite,
    types: details.types,
    weaknesses,
    evolutions: evolutionChain,
    abilities: details.abilities,
    genderRatio: {
      male: maleRatio,
      female: femaleRatio
    }
  };
};

export const usePokemonById = (id: number) => {
  return useQuery({
    queryKey: ['pokemon', id],
    queryFn: () => fetchPokemonById(id),
    enabled: !!id // Só busca quando ID for válido
  });
};
