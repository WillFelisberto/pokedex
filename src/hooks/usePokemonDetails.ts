/* eslint-disable @typescript-eslint/no-explicit-any */
'use client';

import { useQuery } from '@tanstack/react-query';
import { useParams, useRouter } from 'next/navigation';

const API_URL = process.env.NEXT_PUBLIC_POKEAPI_URL;

const fetchPokemonById = async (id: number) => {
  const response = await fetch(`${API_URL}/pokemon/${id}`);
  if (!response.ok) throw new Error('Pokémon não encontrado');
  const data = await response.json();

  return {
    id: data.id,
    name: data.name,
    sprites: {
      front_default: data.sprites.other['official-artwork'].front_default
    },
    types: data.types.map((t: any) => t.type.name),
    height: data.height,
    weight: data.weight,
    baseStats: {
      hp: data.stats[0].base_stat,
      attack: data.stats[1].base_stat,
      defense: data.stats[2].base_stat,
      specialAttack: data.stats[3].base_stat,
      specialDefense: data.stats[4].base_stat,
      speed: data.stats[5].base_stat
    }
  };
};

export const usePokemonDetails = () => {
  const params = useParams(); // Pegamos o "slug" da URL
  const router = useRouter();

  const currentPage = Number(params.slug) || 1;

  const {
    data: pokemon,
    isFetching,
    error
  } = useQuery({
    queryKey: ['pokemon', currentPage],
    queryFn: () => fetchPokemonById(currentPage),
    staleTime: 5000
  });

  const goToPage = (page: number) => {
    if (page < 1 || page > 1010) return; // Número total de Pokémon na API
    router.push(`/${page}`, { scroll: false });
  };

  return { pokemon, isFetching, error, goToPage, currentPage };
};
