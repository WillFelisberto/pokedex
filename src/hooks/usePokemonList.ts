/* eslint-disable @typescript-eslint/no-explicit-any */
'use client';

import { useQuery } from '@tanstack/react-query';
import { useSearchParams, useRouter } from 'next/navigation';
import { useEffect, useState } from 'react';

import { PokemonType } from '@/lib/utils/poketypes';

const API_URL = 'https://pokeapi.co/api/v2';
const POKEMONS_PER_PAGE = 12;

const fetchPokemons = async (page: number) => {
  const offset = (page - 1) * POKEMONS_PER_PAGE;
  const res = await fetch(`${API_URL}/pokemon?offset=${offset}&limit=${POKEMONS_PER_PAGE}`);
  const data = await res.json();

  const detailedPokemons = await Promise.all(
    data.results.map(async (pokemon: { url: string }) => {
      const res = await fetch(pokemon.url);
      const details = await res.json();

      return {
        id: details.id,
        name: details.name,
        sprites: {
          front_default: details.sprites.other['official-artwork'].front_default
        },
        types: details.types.map((t: any) => t.type.name) as PokemonType[],
        height: details.height,
        weight: details.weight
      };
    })
  );

  return { pokemons: detailedPokemons, total: data.count };
};

const searchPokemon = async (query: string) => {
  const res = await fetch(`${API_URL}/pokemon/${query.toLowerCase()}`);
  if (!res.ok) return null;
  const details = await res.json();

  return [
    {
      id: details.id,
      name: details.name,
      sprites: {
        front_default: details.sprites.other['official-artwork'].front_default
      },
      types: details.types.map((t: any) => t.type.name) as PokemonType[],
      height: details.height,
      weight: details.weight
    }
  ];
};

export const usePokemonList = () => {
  const searchParams = useSearchParams();
  const router = useRouter();

  const initialPage = Number(searchParams.get('page')) || 1;
  const [currentPage, setCurrentPage] = useState(initialPage);
  const [searchQuery, setSearchQuery] = useState('');
  const [totalPages, setTotalPages] = useState<number | null>(null);

  const { data, isFetching, refetch, error } = useQuery({
    queryKey: ['pokemonList', currentPage],
    queryFn: () => fetchPokemons(currentPage),
    enabled: !searchQuery,
    staleTime: 5000
  });

  const { data: searchedPokemon, refetch: refetchSearch } = useQuery({
    queryKey: ['searchPokemon', searchQuery],
    queryFn: () => searchPokemon(searchQuery),
    enabled: !!searchQuery,
    staleTime: 5000
  });

  useEffect(() => {
    if (data?.total) {
      setTotalPages(Math.ceil(data.total / POKEMONS_PER_PAGE));
    }
  }, [data]);

  useEffect(() => {
    if (searchQuery) {
      refetchSearch();
    } else {
      refetch();
    }
  }, [searchQuery, currentPage, refetch, refetchSearch]);

  const goToPage = (page: number) => {
    if (page > 0 && totalPages && page <= totalPages) {
      setCurrentPage(page);
      router.push(`?page=${page}`, { scroll: false });
    }
  };

  return {
    pokemons: searchQuery ? searchedPokemon || [] : data?.pokemons || [],
    isFetching,
    error,
    goToPage,
    currentPage,
    setSearchQuery,
    totalPages
  };
};
