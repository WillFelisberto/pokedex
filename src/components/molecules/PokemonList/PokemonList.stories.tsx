import { mockPokemon } from '@/mocks/pokemonMock';
import type { Meta, StoryObj } from '@storybook/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import React from 'react';

import { PokeCard } from '@/components/organisms/PokeCard';

const MockedPokemonList = () => {
  const pokemons = [mockPokemon];
  const isFetching = false;
  const loadMore = () => console.warn('Load More clicked');

  return (
    <div className="flex flex-col items-center p-4">
      <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-3 gap-[10px]">
        {pokemons.map((pokemon) => (
          <PokeCard key={pokemon.id} pokemon={pokemon} />
        ))}
      </div>
      <button
        onClick={loadMore}
        disabled={isFetching}
        className="mt-4 px-4 py-2 bg-blue-500 text-black rounded-lg disabled:opacity-50"
      >
        {isFetching ? 'Carregando...' : 'Carregar Mais'}
      </button>
    </div>
  );
};

const queryClient = new QueryClient();

const meta: Meta<typeof MockedPokemonList> = {
  title: 'Organisms/PokemonList',
  component: MockedPokemonList,
  decorators: [
    (Story) => (
      <QueryClientProvider client={queryClient}>
        <div style={{ padding: '2rem', maxWidth: '1200px', margin: '0 auto' }}>
          <Story />
        </div>
      </QueryClientProvider>
    )
  ]
};

export default meta;

type Story = StoryObj<typeof MockedPokemonList>;

export const Default: Story = {};
