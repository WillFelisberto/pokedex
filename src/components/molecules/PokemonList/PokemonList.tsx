'use client';

import { usePokemonList } from '@/hooks/usePokemonList';
import React from 'react';

import { Loader } from '@/components/atoms/Loader';
import { Pagination } from '@/components/atoms/Pagination';
import { PokeCard } from '@/components/organisms/PokeCard';

export const PokemonList = () => {
  const { pokemons, isFetching, totalPages, currentPage, goToPage } = usePokemonList();

  return (
    <div className="flex flex-col items-center p-4">
      {isFetching && <Loader />}

      <div className="flex flex-wrap justify-center gap-[1.5rem] w-full max-w-[1440px] px-4 sm:px-0">
        {pokemons.map((pokemon) => (
          <PokeCard key={pokemon.id} pokemon={pokemon} />
        ))}
      </div>

      <Pagination
        currentPage={currentPage}
        totalPages={totalPages}
        goToPage={goToPage}
        isFetching={isFetching}
      />
    </div>
  );
};
