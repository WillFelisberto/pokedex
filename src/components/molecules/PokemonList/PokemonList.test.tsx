import { usePokemonList } from '@/hooks/usePokemonList';
import { mockPokemon } from '@/mocks/pokemonMock';
import { render, screen, fireEvent } from '@testing-library/react';
import React from 'react';

import { PokemonList } from './PokemonList';

jest.mock('@/hooks/usePokemonList');

const mockUsePokemonList = usePokemonList as jest.MockedFunction<typeof usePokemonList>;

const mockPokemons = [mockPokemon];

describe('<PokemonList />', () => {
  beforeEach(() => {
    mockUsePokemonList.mockReturnValue({
      pokemons: mockPokemons,
      isFetching: false,
      currentPage: 1,
      setSearchQuery: jest.fn(),
      error: null,
      totalPages: 10,
      goToPage: jest.fn()
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render the list of pokemons', () => {
    const { container } = render(<PokemonList />);

    mockPokemons.forEach((pokemon) => {
      const pokemonName = screen.getByText(pokemon.name);
      expect(pokemonName).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });

  it('should render the loader when isFetching is true', () => {
    mockUsePokemonList.mockReturnValue({
      pokemons: [],
      isFetching: true,
      currentPage: 1,
      totalPages: 10,
      setSearchQuery: jest.fn(),
      error: null,
      goToPage: jest.fn()
    });

    render(<PokemonList />);

    expect(screen.getByTestId('loader')).toBeInTheDocument();
  });

  it('should render pagination', () => {
    render(<PokemonList />);

    expect(screen.getByText('1')).toBeInTheDocument();
    expect(screen.getByText('→')).toBeInTheDocument();
  });

  it('should disable pagination buttons when fetching', () => {
    mockUsePokemonList.mockReturnValue({
      pokemons: [],
      isFetching: true,
      currentPage: 1,
      totalPages: 10,
      setSearchQuery: jest.fn(),
      error: null,
      goToPage: jest.fn()
    });

    render(<PokemonList />);

    expect(screen.getByText('←')).toBeDisabled();
    expect(screen.getByText('→')).toBeDisabled();
  });

  it('should call goToPage when clicking on pagination buttons', () => {
    render(<PokemonList />);

    const nextButton = screen.getByText('→');
    fireEvent.click(nextButton);

    expect(mockUsePokemonList().goToPage).toHaveBeenCalledWith(2);
  });

  it('should disable the previous button on the first page', () => {
    mockUsePokemonList.mockReturnValue({
      pokemons: [],
      isFetching: false,
      currentPage: 1,
      totalPages: 10,
      setSearchQuery: jest.fn(),
      error: null,
      goToPage: jest.fn()
    });

    render(<PokemonList />);

    expect(screen.getByText('←')).toBeDisabled();
  });

  it('should disable the next button on the last page', () => {
    mockUsePokemonList.mockReturnValue({
      pokemons: [],
      isFetching: false,
      currentPage: 10,
      totalPages: 10,
      setSearchQuery: jest.fn(),
      error: null,
      goToPage: jest.fn()
    });

    render(<PokemonList />);

    expect(screen.getByText('→')).toBeDisabled();
  });
});
