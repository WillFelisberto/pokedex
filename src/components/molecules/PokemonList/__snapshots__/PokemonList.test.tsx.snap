// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonList /> should render the list of pokemons 1`] = `
<div>
  <div
    class="flex flex-col items-center p-4"
  >
    <div
      class="flex flex-wrap justify-center gap-[1.5rem] w-full max-w-[1440px] px-4 sm:px-0"
    >
      <div
        class="bg-white min-w-[340px] max-w-[340px] p-2 rounded-lg text-center flex flex-col gap-4"
        data-testid="poke-card"
      >
        <a
          href="/pokemon/2"
        >
          <div
            class="relative w-full h-[280px] flex flex-col items-center justify-center rounded-b-3xl overflow-hidden"
          >
            <div
              class="relative w-full h-full flex items-center justify-center"
              data-testid="pokemon-header"
              style="clip-path: ellipse(60% 65% at 50% 30%);"
            >
              <div
                class="absolute inset-0 flex items-center justify-center"
              >
                <div
                  class="absolute inset-0 w-1/2 bg-gradient-to-l from-white/15 to-transparent"
                />
                <div
                  class="relative"
                >
                  <img
                    alt="grass"
                    height="140"
                    src="/icons/png/filled/grass.png"
                    style="mask-image: linear-gradient(135deg, rgba(0 0 0 / 50%), rgba(0,0,0,0)); filter: brightness(10);"
                    width="140"
                  />
                </div>
              </div>
            </div>
            <div
              class="  w-full flex justify-center mt-[5rem] absolute"
            >
              <img
                alt="ivysaur"
                class="z-10"
                height="215"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/2.png"
                width="215"
              />
            </div>
          </div>
          <div
            class="p-2 text-left"
          >
            <h2
              class="text-black font-bold capitalize text-3xl"
              data-testid="pokemon-name"
            >
              ivysaur
            </h2>
            <p
              class="text-base text-gray-700"
              data-testid="pokemon-id"
            >
              Nº002
            </p>
          </div>
          <div
            class="p-2 flex gap-2"
            data-testid="pokemon-types"
          >
            <div
              aria-label="grass"
              class="flex w-fit items-center gap-2 text-white px-3 py-2 rounded-full bg-pokemon-grass"
              data-testid="poke-badge"
            >
              <div
                class="w-6 h-6 flex items-center justify-center rounded-full overflow-visible bg-white"
              >
                <img
                  alt="grass"
                  height="13"
                  src="/icons/png/filled/grass.png"
                  width="13"
                />
              </div>
              <span
                class="font-semibold text-sm"
              >
                Grass
              </span>
            </div>
            <div
              aria-label="poison"
              class="flex w-fit items-center gap-2 text-white px-3 py-2 rounded-full bg-pokemon-poison"
              data-testid="poke-badge"
            >
              <div
                class="w-6 h-6 flex items-center justify-center rounded-full overflow-visible bg-white"
              >
                <img
                  alt="poison"
                  height="13"
                  src="/icons/png/filled/poison.png"
                  width="13"
                />
              </div>
              <span
                class="font-semibold text-sm"
              >
                Poison
              </span>
            </div>
          </div>
          <div
            class="p-2 w-full flex gap-[20px]"
          >
            <div
              class="grid grid-cols-2 w-full gap-[20px]"
            >
              <div
                class="flex flex-col gap-[4.5px] items-left   rounded-lg  "
              >
                <div
                  class="flex items-end gap-[8.5px] h-[16px] text-gray-500 text-xs font-semibold uppercase"
                >
                  <img
                    alt="weight"
                    height="18"
                    src="/icons/svg/outlined/weight.svg"
                    width="18"
                  />
                  <span>
                    Peso
                  </span>
                </div>
                <div
                  class="mt-1 font-(--font-poppins) h-[42px]  w-full py-[8px] bg-white rounded-[15px] text-gray-800 font-bold text-lg text-center border border-[#E5E5E5]"
                >
                  13 kg
                </div>
              </div>
              <div
                class="flex flex-col gap-[4.5px] items-left   rounded-lg  "
              >
                <div
                  class="flex items-end gap-[8.5px] h-[16px] text-gray-500 text-xs font-semibold uppercase"
                >
                  <img
                    alt="height"
                    height="15"
                    src="/icons/svg/outlined/height.svg"
                    width="15"
                  />
                  <span>
                    Altura
                  </span>
                </div>
                <div
                  class="mt-1 font-(--font-poppins) h-[42px]  w-full py-[8px] bg-white rounded-[15px] text-gray-800 font-bold text-lg text-center border border-[#E5E5E5]"
                >
                  1 m
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
    <div
      class="flex items-center gap-2 py-4"
    >
      <button
        class="px-4 py-2 bg-gray-700 text-white rounded-md disabled:opacity-50 disabled:cursor-not-allowed hover:bg-gray-600 transition"
        disabled=""
      >
        ←
      </button>
      <button
        class="px-4 py-2 rounded-md transition bg-blue-600 text-white font-bold disabled:opacity-50 disabled:cursor-not-allowed"
      >
        1
      </button>
      <button
        class="px-4 py-2 rounded-md transition bg-gray-700 text-white hover:bg-gray-600 disabled:opacity-50 disabled:cursor-not-allowed"
      >
        2
      </button>
      <button
        class="px-4 py-2 rounded-md transition bg-gray-700 text-white hover:bg-gray-600 disabled:opacity-50 disabled:cursor-not-allowed"
      >
        3
      </button>
      <button
        class="px-4 py-2 rounded-md transition bg-gray-700 text-white hover:bg-gray-600 disabled:opacity-50 disabled:cursor-not-allowed"
      >
        4
      </button>
      <button
        class="px-4 py-2 rounded-md transition bg-gray-700 text-white hover:bg-gray-600 disabled:opacity-50 disabled:cursor-not-allowed"
      >
        5
      </button>
      <button
        class="px-4 py-2 bg-gray-700 text-white rounded-md disabled:opacity-50 disabled:cursor-not-allowed hover:bg-gray-600 transition"
      >
        →
      </button>
    </div>
  </div>
</div>
`;
